pkg/config/remote_config.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/prometheus/common/config"
+)
+
+// supported remote config provider schemes
+const (
+	HTTP  = "http"
+	HTTPS = "https"
+	// TODO: add s3, gcs, blob, and git providers backed by go-fsimpl
+)
+
+// RemoteOpts struct contains agent remote config options
+type RemoteOpts struct {
+	URL              *url.URL
+	HTTPClientConfig *config.HTTPClientConfig
+}
+
+// RemoteProvider ...
+type RemoteProvider interface {
+	Retrieve() ([]byte, error)
```

Comment:
in the future, we should allow this to have a timeout

Reply:
Agreed. For now, if we want to set a timeout for HTTP/S, we can pass the `WithIdleConnTimeout` `HTTPClientOption` to `NewClientFromConfig`.  We can revisit managing our own timeouts once we have more providers to deal with.